notify/webex/webex.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webex
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	commoncfg "github.com/prometheus/common/config"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	// maxMessageSize represents the maximum message length that Webex supports.
+	maxMessageSize = 7439
+)
+
+type Notifier struct {
+	conf    *config.WebexConfig
+	tmpl    *template.Template
+	logger  log.Logger
+	client  *http.Client
+	retrier *notify.Retrier
+	APIURL  *config.URL
+}
+
+// New returns a new Webex notifier.
+func New(c *config.WebexConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	if c.HTTPConfig.Authorization == nil && c.HTTPConfig.BearerToken == "" {
+		c.HTTPConfig.Authorization = &commoncfg.Authorization{
+			Type:        "Bearer",
+			Credentials: commoncfg.Secret(c.BotToken),
+		}
+	}
```

Comment:
I thought about it but it's very unintuitive if you ask me and leads to a poor UX (You have 100% read the manual to get the correct configuration) - Other similar integrations (such as Telegram) only use the explicit version.

If you don't have a strong opinion I would rather allow both (use what you know if you're already familiar with how the HTTP config works) or have an easy way to know what do you need to get this integration to work.

Reply:
> Telegram is different because the token isn't passed via HTTP header (instead it's part of the URL)

I was thinking about OpsGenie, sorry about the confusion.

> Also the existing HTTP client config can load the token from file + reload the content if it changes so I'd assume that folks will ask for bot_token_file too

That's a good shout. I don't feel too strongly about this - so I'll remove the `bot_token` then.

Reply:
> I was thinking about OpsGenie, sorry about the confusion.

Right, `api_key` exists because the OpsGenie integration predates the ability to configure the Authorization header with an authentication type other than `Bearer`. But something to keep in mind when reviewing new integrations.